ignite/pkg/xast/function.go

```diff
@@ -122,204 +168,389 @@ func AppendFuncStruct(name, param, code string, index int) FunctionOptions {
 	}
 }
 
-// NewFuncReturn replaces return statements in a Go function with a new return statement.
+// NewFuncReturn replaces return statements in a function.
 func NewFuncReturn(returnVars ...string) FunctionOptions {
 	return func(c *functionOpts) {
 		c.returnVars = append(c.returnVars, returnVars...)
 	}
 }
 
-func newFunctionOptions() functionOpts {
+// newFunctionOptions creates a new functionOpts with defaults.
+func newFunctionOptions(funcName string) functionOpts {
 	return functionOpts{
+		funcName:       funcName,
 		newParams:      make([]functionParam, 0),
 		body:           "",
 		newLines:       make([]functionLine, 0),
-		insideCall:     make([]functionCall, 0),
-		insideStruct:   make([]functionStruct, 0),
+		insideCall:     make(functionCalls, 0),
+		insideStruct:   make(functionStructs, 0),
 		appendTestCase: make([]string, 0),
 		appendCode:     make([]string, 0),
 		returnVars:     make([]string, 0),
 	}
 }
 
-// ModifyFunction modify a function based in the options.
-func ModifyFunction(fileContent, functionName string, functions ...FunctionOptions) (modifiedContent string, err error) {
-	// Apply function options.
-	opts := newFunctionOptions()
-	for _, o := range functions {
-		o(&opts)
+// ModifyFunction modifies a function in Go source code using functional options.
+func ModifyFunction(content string, funcName string, functions ...FunctionOptions) (string, error) {
+	// Collect all function options.
+	opts := newFunctionOptions(funcName)
+	for _, fn := range functions {
+		fn(&opts)
 	}
 
-	fileSet := token.NewFileSet()
-
-	// Parse the Go source code content.
-	f, err := parser.ParseFile(fileSet, "", fileContent, parser.ParseComments)
+	// Parse source into AST
+	fset := token.NewFileSet()
+	file, err := parser.ParseFile(fset, "", content, parser.ParseComments)
 	if err != nil {
+		return "", errors.Errorf("failed to parse file: %w", err)
+	}
+
+	// Find the target function.
+	funcDecl := findFuncDecl(file, funcName)
+	if funcDecl == nil {
+		return "", errors.Errorf("function %q not found", funcName)
+	}
+
+	// Apply modifications.
+	if err := applyFunctionOptions(fset, funcDecl, &opts); err != nil {
 		return "", err
 	}
 
-	// Parse the content of the new function into an ast.
-	var newFunctionBody *ast.BlockStmt
-	if opts.body != "" {
-		newFunctionBody, err = codeToBlockStmt(fileSet, opts.body)
-		if err != nil {
-			return "", err
+	// Format and return the modified source.
+	var buf bytes.Buffer
+	if err := format.Node(&buf, fset, file); err != nil {
+		return "", errors.Errorf("failed to format modified file: %w", err)
+	}
+	return formatNode(fset, file)
+}
+
+// findFuncDecl finds a function declaration in an AST by name.
+func findFuncDecl(file *ast.File, name string) *ast.FuncDecl {
```

Comment:
This is new, what does it adds?

Reply:
this tries to find a function inside a file by the function name, and returns the `*ast.FuncDecl`